common/test/src/ui/tst_MapDocument.cpp

```diff
@@ -462,6 +463,63 @@ TEST_CASE_METHOD(MapDocumentTest, "canUpdateLinkedGroups")
     document->canUpdateLinkedGroups(kdl::vec_static_cast<mdl::Node*>(entityNodes)));
 }
 
+TEST_CASE_METHOD(MapDocumentTest, "newDocument")
+{
+  const auto fgdContent = R"(
+    @SolidClass = worldspawn : "World entity" [
+      some_default_prop(string) : "" : "value"
+    ]
+  )";
+
+  const auto configContent = R"(
+    {
+        version: 9,
+        name: "testNewDocument",
+        icon: "Icon.png",
+        "fileformats": [ { "format": "Standard" } ],
+        "filesystem": {
+            "searchpath": ".",
+            "packageformat": { "extension": ".pak", "format": "idpak" }
+        },
+        "materials": {
+            "root": "textures",
+            "format": { "extensions": [".jpg"], "format": "image" }
+        },
+        "entities": {
+            "definitions": [ "WorldSpawnOnly.fgd" ],
+            "defaultcolor": "0.6 0.6 0.6 1.0",
+            "setDefaultProperties": true
+        },
+        "tags": { "brush": [], "brushface": [] },
+        "faceattribs": { "surfaceflags": [], "contentflags": [] }
+    }
+  )";
+
+  SECTION("Default worldspawn properties")
+  {
+    const auto gameName = "testNewDocument";
+    const auto dirPath = std::filesystem::current_path() / "fixture/games" / gameName;
+
+    io::TestEnvironment{dirPath.string(), [&](auto& env) {
```

Comment:
If you only create a temporary instance of TestEnvironment, the files will be deleted in the destructor. I'm surprised that this test works at all. You also shouldn't pass an absolute path into the constructor. For your purposes, you can pass an empty path here. `TestEnvironment` creates a temporary sandbox, and the given path will just be appended to that.

````
auto env = TestEnvironment{std::filesystem::path{}, [](auto& env) {
...
}};

...
```

Reply:
Didn't know about that, I guess the compiler was smarter in this case, thanks!